server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -647,22 +678,53 @@ private void stopBeingLeader()
     return ImmutableList.of(compactSegments);
   }
 
-  private class DutiesRunnable implements Runnable
+  protected class DutiesRunnable implements Runnable
```

Comment:
I see DutiesRunnable used to be public with a protected constructor but was flipped to private/private in [pull 9644](https://github.com/apache/druid/pull/9644) ... Now we are flipping the class and constructor to protected. I want to make sure @maytasm sees this PR and can comment on the idea of going back to protected from private.

Reply:
LGTM. Maybe add @VisibleForTesting too

Reply:
updated